src/devices/Seesaw/SeesawBase.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+
+using System.Device.I2c;
+
+namespace Iot.Device.Seesaw
+{
+    public partial class Seesaw : IDisposable
+    {
+        private const byte Sessaw_Hardware_Id = 0x55;
+
+        public I2cDevice I2cDevice {get; private set;}
+        private UInt32 _options;
+
+        /// <summary>
+        /// Initializes new instance of Seesaw.
+        /// </summary>
+        /// <param name="i2cDevice">The I2cDevice to be used to communicate with the SeeSaw module. Note that the I2cDevice
+        /// will be disposed when the along with the SeeSaw device</param>
+        public Seesaw(I2cDevice i2cDevice)
+        {
+            Initialize(i2cDevice);
+        }
+
+        /// <summary>
+        /// Version of the SeeSaw module. 
+        /// </summary>
+        public UInt32 Version { get; private set; }
+
+        /// <summary>
+        /// Reads the temperature of the SeeSaw device. 
+        /// </summary>
+        /// <returns>A float that represents the temperature in degrees celcius.</returns>
+        public float GetTemperature()
+        {
+            byte[] buffer = Read(Seesaw_Module.Status, Seesaw_Function.Status_Temp, 4, 1000);
+
+            return ((1.0F / (1UL << 16)) * (((UInt32)buffer[0] << 24) | ((UInt32)buffer[1] << 16) | ((UInt32)buffer[2] << 8) | (UInt32)buffer[3]));
```

Comment:
If powwible use https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1

Reply:
Done in the latest PR update.